network/websocket_dataframe.ksy

```diff
@@ -0,0 +1,68 @@
+meta:
+  id: websocket_dataframe
+  title: WebSocket
+  xref:
+    rfc: 6455
+  endian: be
+  license: APACHE-2.0
+doc: |
+  The WebSocket protocol establishes a two-way communication channel via TCP.
+  Messages are made up of one or more dataframes, and are delineated by
+  frames with the `fin` bit set.
+seq:
+  - id: fin
+    type: b1
+  - id: rsv1
+    type: b1
+  - id: rsv2
+    type: b1
+  - id: rsv3
+    type: b1
+  - id: opcode
+    enum: opcode
+    type: b4
+  - id: b1
+    type: u1
+  - id: extended_len_1
+    type: u2
+    if: payload_len == 126
+  - id: extended_len_2
+    type: u4
+    if: payload_len == 127
+  - id: mask_key
+    type: u4
+    if: is_masked == 1
+  - id: payload
+    size: len
+    
+instances:
+  is_masked:
+    value: (b1 & 0b10000000) >> 7
+  payload_len:
+    value: (b1 & 0b01111111)
```

Comment:
The only way `b1` is used is to split up into these two, so it would make sense to implement is using `bX` as well:

```yaml
seq:
  # ...
  - id: is_masked
    type: b1
  - id: len_payload
    type: b7
```

Also please note the [style guide](http://doc.kaitai.io/ksy_style_guide.html) — the general guidance is to use `len_XXX` identifiers for attributes that carry size information for `XXX` attribute.

Reply:
Happy to change the names, but there were issues with `if: is_masked == 1` being used to either include the masking bits or not. Given that, I opted to leave it as a byte that gets parsed later. If there's a better solution, let me know.

Reply:
They're all still independent attributes? I mean, why don't we just do:

```yaml
  - id: is_masked
    type: b1
  - id: len_payload_primary
    type: b7
  - id: len_payload_extended_1
    type: u2
    if: len_payload_primary == 126
  - id: len_payload_extended_2
    type: u4
    if: len_payload_primary == 127
  - id: mask_key
    type: u4
    if: is_masked
```

Reply:
That works super well! I hadn't thought of it at the time, but it's now done.